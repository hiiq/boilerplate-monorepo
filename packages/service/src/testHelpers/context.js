import { User, Utils } from '@boilerplate-monorepo/common';

const stubLoader = name => ({
  clear: td.func(`${name}.clear`),
  load: td.func(`${name}.load`),
});

const db = {
  user: {
    read: jest.fn().mockName('read'),
    readRaw: jest.fn().mockName('readRaw'),
    recoveryFind: jest.fn().mockName('recoveryFind'),
    register: jest.fn().mockName('register'),
    revokeRefreshTokens: jest.fn().mockName('revokeRefreshTokens'),
    save: jest.fn().mockName('save'),
  },
  users: {
    read: jest.fn().mockName('read'),
    readPaginated: jest.fn().mockName('readPaginated'),
    readRaw: jest.fn().mockName('readRaw'),
  },
};

const make = Utils.makeSafeExample({
  db,
  ip: '127.0.0.1',
  loaders: {
    user: stubLoader('user'),
  },
  req: {
    headers: {},
    ip: '127.0.0.1',
  },
  res: {
    cookie: jest.fn().mockName('cookie'),
  },
  user: User.apiExample(),
});

export { make };
