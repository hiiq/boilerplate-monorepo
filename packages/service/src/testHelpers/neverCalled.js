const neverCalled = { ignoreExtraArgs: true, times: 0 };

export { neverCalled };
