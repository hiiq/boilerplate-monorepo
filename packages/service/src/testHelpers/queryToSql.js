import { pick } from 'ramda';

const queryToSql = pick(['bindings', 'sql', 'transaction']);

export { queryToSql };
