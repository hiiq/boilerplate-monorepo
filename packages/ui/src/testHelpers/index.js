import { cleanup, render } from '@testing-library/react';
import { curry } from 'ramda';
import React from 'react';
import { Providers } from 'modules/app';

export * from '@testing-library/react';

const customRender = curry((Component, defaultProps = {}, overrides = {}) =>
  render(<Component {...defaultProps} {...overrides} />, {
    wrapper: Providers,
  })
);

export { cleanup, customRender as render };
