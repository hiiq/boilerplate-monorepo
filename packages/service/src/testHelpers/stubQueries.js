import knexMock from 'mock-knex';
import { queryToSql } from './queryToSql';

const stubQueries = (mutateFn, responses = []) => {
  const tracker = knexMock.getTracker();

  tracker.install();

  let queries = [];

  tracker.on('query', (query, step) => {
    queries = [...queries, queryToSql(query)];
    mutateFn(queries);
    query.response(responses[step - 1]);
  });
};

export { stubQueries };
