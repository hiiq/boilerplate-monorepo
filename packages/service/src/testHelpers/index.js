import { make as makeContext } from './context';
import {
  graphqlClient,
  makeGraphqlClient,
  responseToData,
  responseToErrorCode,
} from './graphql';
import { neverCalled } from './neverCalled';
import { queryToSql } from './queryToSql';
import { stubQueries } from './stubQueries';

export {
  graphqlClient,
  makeContext,
  makeGraphqlClient,
  neverCalled,
  queryToSql,
  responseToData,
  responseToErrorCode,
  stubQueries,
};
