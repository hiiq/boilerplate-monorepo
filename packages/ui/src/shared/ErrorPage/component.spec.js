import React from 'react';
import { render } from 'testHelpers';
import { ErrorPage, domTestId } from '.';

describe('<ErrorPage/>', () => {
  const defaultProps = {
    message: 'MESSAGE',
    onFeedbackClick: jest.fn().mockName('onFeedbackClick'),
  };

  it('renders in production properly', () => {
    const { getByTestId } = render(<ErrorPage {...defaultProps} />);
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });

  it('should not render the error stack in production', () => {
    const { queryByTestId } = render(<ErrorPage {...defaultProps} />);
    const snapshot = queryByTestId('errorDetails');

    expect(snapshot).toBeNull();
  });
});
