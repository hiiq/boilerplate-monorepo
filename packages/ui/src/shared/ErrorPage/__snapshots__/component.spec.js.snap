// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage/> renders in production properly 1`] = `
<div
  class="Content__Styled-sc-1ovcgwr-0 ginwwh"
  data-testid="ErrorPage"
>
  <div
    class="Header__Styled-sc-1wj09bb-0 kDHlTu"
    data-testid="Header"
  >
    <div
      data-autofocus-inside="true"
    >
      <h1
        class="Header__H1-sc-1wj09bb-1 dbepyo"
        id="main-content"
        tabindex="0"
      >
        t(title)
      </h1>
    </div>
  </div>
  <section
    class="ErrorPage__Styled-y0bbzq-0 bQBAQL Body__Styled-sc-4524au-0 cnhjhz"
    data-testid="Body"
  >
    <div />
    <div
      class="ErrorPage__Center-y0bbzq-1 kRztOy"
    >
      <img
        alt="t(officePanda)"
        src="officePanda.gif"
      />
      <p>
        t(body)
      </p>
      <button
        aria-disabled="false"
        aria-label="t(submitFeedback)"
        class="Button__Styled-sc-13kes55-0 dVNaCg"
        data-testid="Button"
        type="button"
      >
        t(submitFeedback)
      </button>
    </div>
  </section>
  <footer
    class="Footer__Styled-sc-18g9rsu-0 jRwNDo"
    data-testid="Footer"
  >
    <span>
      t(footer)
    </span>
    <div />
    <section
      class="Toggles__Styled-pxa8zq-0 AJWad"
      data-testid="Toggles"
    >
      <button
        aria-disabled="false"
        aria-label="t(modes.dark)"
        class="ToggleButton__ProxyButton-sc-1lul5b-0 etiKNI Button__Styled-sc-13kes55-0 UmzYp"
        data-testid="ToggleTheme"
        type="button"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
          />
        </svg>
      </button>
    </section>
  </footer>
</div>
`;
